userspace/libscap/scap_savefile.c

```diff
@@ -1297,6 +1300,7 @@ static int32_t scap_read_proclist(scap_t *handle, scap_reader_t* r, uint32_t blo
 		case PL_BLOCK_TYPE_V9:
 			readsize = scap_reader_read(r, &(sub_len), sizeof(uint32_t));
 			CHECK_READ_SIZE(readsize, sizeof(uint32_t));
+			scap_reader_be32toh(r, &(sub_len));

```

Comment:
All these feel like we want a `scap_reader_read_u{8,16,32,64}` family of functions. Sadly, `CHECK_READ_SIZE` has a `return` inside so it can't be a function, but we might be able to clean this up somehow anyway.